sympy/physics/mechanics/kane.py

```diff
@@ -227,19 +246,25 @@ def __init__(self, frame, q_ind, u_ind, kd_eqs=None, q_dependent=None,
         self._forcelist = forcelist
         self._bodylist = bodies
 
+        self._lin_vel_constr = velocity_constraints
+        self._nonlin_vel_constr = nonlinear_velocity_constraints
+        self._acc_constraints = acceleration_constraints
+
```

Comment:
```suggestion
```
You don't have to store these. You can also remove it from `_form_fr` and `_form_frstar`. To get those numbers you could consider using `_f_h`, `_f_nh`, etc.

Reply:
You require the number of constraints, not the constraints themselves. You can retrieve the number of linear velocity constraints using `len(self._f_h)`, similar things are available for the other two.

Reply:
I am a little reluctant to change this:
1. It is so in the original code: the self.lin_vel_constr, self._acc_constraints are there, I simply copied it.
2. If feel it makes the code easier to read: e.g. in lines 616 and 621 like this is is clear what is being checked in the if